x/ecocredit/server/invariants_test.go

```diff
@@ -1,371 +1,268 @@
-					RetiredBalance: "100.123",
-				},
-				{
-					Address:        acc2.String(),
-					BatchDenom:     "1/2",
-					RetiredBalance: "210.456",
-				},
-			},
-			[]*ecocredit.Supply{
-				{
-					BatchDenom:     "1/2",
-					RetiredSupply:  "310.579",
-					TradableSupply: "0",
+					BatchId:        1,
+					TradableAmount: "110",
+					RetiredAmount:  "10",
 				},
 				{
-					BatchDenom:     "3/4",
-					RetiredSupply:  "1234",
-					TradableSupply: "0",
-				},
-			},
-			true,
-		},
-		{
-			"fail with error retired supply does not match",
-			[]*ecocredit.Balance{
-				{
-					Address:        acc1.String(),
-					BatchDenom:     "1/2",
-					RetiredBalance: "100.123",
-				},
-				{
-					Address:        acc2.String(),
-					BatchDenom:     "1/2",
-					RetiredBalance: "210.456",
-				},
-				{
-					BatchDenom:     "3/4",
-					Address:        acc2.String(),
-					RetiredBalance: "1234",
-				},
-			},
-			[]*ecocredit.Supply{
-				{
-					BatchDenom:     "1/2",
-					RetiredSupply:  "310.57",
-					TradableSupply: "0",
-				},
-				{
-					BatchDenom:     "3/4",
-					RetiredSupply:  "1234",
-					TradableSupply: "0",
+					BatchId:        2,
+					TradableAmount: "2000",
+					RetiredAmount:  "123",
 				},
 			},
+			map[uint64]math.Dec{},
 			true,
 		},
 	}
 
 	for _, tc := range testCases {
 		tc := tc
-		ctx, storeKey := setupStore(t)
-		store := ctx.KVStore(storeKey)
+		suite := setupBase(t)
 		t.Run(tc.msg, func(t *testing.T) {
-			initBalances(t, store, tc.balances)
-			initSupply(t, store, tc.supply)
+			initBalances(t, suite.ctx, suite.stateStore, tc.balances)
+			initSupply(t, suite.ctx, suite.stateStore, tc.supply)
 
-			msg, broken := retiredSupplyInvariant(store)
+			msg, broken := coreserver.BatchSupplyInvariant(suite.ctx, suite.k, tc.basketBalance)
 			if tc.expBroken {
 				require.True(t, broken, msg)
 			} else {
 				require.False(t, broken, msg)
 			}
 		})
 	}
+
 }
 
-func initBalances(t *testing.T, store sdk.KVStore, balances []*ecocredit.Balance) {
+func initBalances(t *testing.T, ctx context.Context, ss api.StateStore, balances []*core.BatchBalance) {
 	for _, b := range balances {
-		denomT := ecocredit.BatchDenomT(b.BatchDenom)
-		addr, err := sdk.AccAddressFromBech32(b.Address)
+		_, err := math.NewNonNegativeDecFromString(b.Tradable)
 		require.NoError(t, err)
-		if b.TradableBalance != "" {
-			d, err := math.NewNonNegativeDecFromString(b.TradableBalance)
-			require.NoError(t, err)
-			key := ecocredit.TradableBalanceKey(addr, denomT)
-			ecocredit.SetDecimal(store, key, d)
-		}
-		if b.RetiredBalance != "" {
-			d, err := math.NewNonNegativeDecFromString(b.RetiredBalance)
-			require.NoError(t, err)
-			key := ecocredit.RetiredBalanceKey(addr, denomT)
-			ecocredit.SetDecimal(store, key, d)
-		}
+
+		require.NoError(t, ss.BatchBalanceTable().Save(ctx, &api.BatchBalance{
+			Address:  b.Address,
+			BatchId:  b.BatchId,
+			Tradable: b.Tradable,
+			Retired:  b.Retired,
+			Escrowed: b.Escrowed,
+		}))
```

Comment:
I think we want to use `Insert` here rather than `Save`. `Save` overwrites an existing entry if it already exists but `Insert` will error if there is an existing entry, which we are not expecting.